### PR TITLE
DTGB-549: Added CTA block fixes.

### PR DESCRIPTION
**Pull request by meirege**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the style guide CHANGELOG accordingly.
- [ ] I have updated the gent_base CHANGELOG accordingly.

### CODE REVIEW

File: gent_base.theme [hunk 1]
```diff
@@ -733,3 +734,56 @@ function gent_base_preprocess_pager(&$variables) {
...
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function gent_base_preprocess_paragraph__call_to_action(&$variables) {
+  $variables['primary_document_size'] = format_size(_gent_base_get_entity_reference_file_size($variables, 'field_primary_action_document'));
+  $variables['secondary_document_size'] = format_size(_gent_base_get_entity_reference_file_size($variables, 'field_secondary_action_document'));
+}
+
+/**
+ * Helper function to get the file size of a referenced File entity.
```
Comment (delrueba) [inline gent_base.theme]: There should be a helper file for these.

File: templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig [hunk 2]
```diff
@@ -31,7 +31,8 @@
           {% endif %}
           {% if primary_type == 'document' %}
             {% if content.field_primary_action_document|render|striptags|trim is not empty %}
-              {{ content.field_primary_action_document }}
+              <a href="{{ file_url(content.field_primary_action_document[0]['#media'].field_media_file.entity.uri.value) }}" class="standalone-link button" download title="{{ content.field_primary_action_title|render|striptags|trim }}">{{ content.field_primary_action_title|render|striptags|trim }}</a>
```
Comment (delrueba) [inline templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig]: Title seems useless, it's identical to the link content.

File: templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig [hunk 3]
```diff
@@ -41,7 +42,8 @@
           {% endif %}
           {% if secondary_type == 'document' %}
             {% if content.field_secondary_action_document|render|striptags|trim is not empty %}
-              {{ content.field_secondary_action_document }}
+              <a href="{{ file_url(content.field_secondary_action_document[0]['#media'].field_media_file.entity.uri.value) }}" class="standalone-link" download title="{{ content.field_secondary_action_title|render|striptags|trim }}">{{ content.field_secondary_action_title|render|striptags|trim }}</a>
```
Comment (delrueba) [inline templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig]: same comment as above

**Review by delrueba — CHANGES_REQUESTED** [review]

File: gent_base.theme [hunk 4]
```diff
@@ -733,3 +734,56 @@ function gent_base_preprocess_pager(&$variables) {
   $variables['items']['first']['text'] = '1';
   $variables['items']['last']['text'] = $total;
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function gent_base_preprocess_paragraph__call_to_action(&$variables) {
+  $variables['primary_document_size'] = format_size(_gent_base_get_entity_reference_file_size($variables, 'field_primary_action_document'));
```
Comment (zero2one) [inline gent_base.theme]: How will this be displayed if the `_gent_base_get_entity_reference_file_size()` function returns NULL?

File: templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig (hunk 2 above)
Comment (zero2one) [inline templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig]: There is a preprocessor for the CTA paragraph.
Better to extract all the values there and add them as extra theme variables.
Makes it easier to print the variables in the template ;-)

**Review by zero2one — CHANGES_REQUESTED** [review]

File: gent_base.theme (hunk 4 above)
Comment (meirege) [reply gent_base.theme]: Twig gaat een check doen op de variables die nodig zijn.

File: templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig (hunk 2 above)
Comment (meirege) [reply templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig]: Juist ja :) toegepast voor de download links.

File: gent_base.theme (hunk 1 above)
Comment (meirege) [reply gent_base.theme]: Fixed.

File: templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig (hunk 2 above)
Comment (meirege) [reply templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig]: Fixed.

File: templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig (hunk 3 above)
Comment (meirege) [reply templates/contrib/paragraphs/cta/paragraph--call-to-action.html.twig]: Fixed.

**Review by delrueba — APPROVED** [review]

**Comment by meirege** [comment]
@zero2one Zijn de changes ok voor jou?

**Review by zero2one — APPROVED** [review]

**Pull request merged**